cmd/kube-controller-manager/app/names/controller_names.go

```diff
@@ -0,0 +1,137 @@
+//     2.1. [TODO] logger component should be configured with the controller name by calling LoggerWithName
+//     2.2. [TODO] logging should use a canonical controller name when referencing a controller (Eg. Starting X, Shutting down X)
+//     2.3. [TODO] emitted events should have an EventSource.Component set to the controller name (usually when initializing an EventRecorder)
+//     2.4. [TODO] registering ControllerManagerMetrics with ControllerStarted and ControllerStopped
+//     2.5. [TODO] calling WaitForNamedCacheSync
+//  3. defining controller options for "--help" command or generated documentation
+//     3.1. controller name should be used to create a pflag.FlagSet when registering controller options (the name is rendered in a controller flag group header)
+//     3.2. when defined flag's help mentions a controller name
+//  4. defining a new service account for a new controller (old controllers may have inconsistent service accounts to stay backwards compatible)
+const (
+	ServiceAccountTokenController                = "serviceaccount-token-controller"
+	EndpointController                           = "endpoint-controller"
+	EndpointSliceController                      = "endpointslice-controller"
+	EndpointSliceMirroringController             = "endpointslice-mirroring-controller"
+	ReplicationControllerController              = "replicationcontroller-controller"
+	PodGarbageCollectorController                = "pod-garbage-collector-controller"
+	ResourceQuotaController                      = "resourcequota-controller"
+	NamespaceController                          = "namespace-controller"
+	ServiceAccountController                     = "serviceaccount-controller"
+	GarbageCollectorController                   = "garbage-collector-controller"
+	DaemonSetController                          = "daemonset-controller"
+	JobController                                = "job-controller"
+	DeploymentController                         = "deployment-controller"
+	ReplicaSetController                         = "replicaset-controller"
+	HorizontalPodAutoscalerController            = "horizontal-pod-autoscaler-controller"
+	DisruptionController                         = "disruption-controller"
+	StatefulSetController                        = "statefulset-controller"
+	CronJobController                            = "cronjob-controller"
+	CertificateSigningRequestSigningController   = "certificatesigningrequest-signing-controller"
+	CertificateSigningRequestApprovingController = "certificatesigningrequest-approving-controller"
+	CertificateSigningRequestCleanerController   = "certificatesigningrequest-cleaner-controller"
+	TTLController                                = "ttl-controller"
+	BootstrapSignerController                    = "bootstrap-signer-controller"
+	TokenCleanerController                       = "token-cleaner-controller"
+	NodeIpamController                           = "node-ipam-controller"
```

Comment:
Here's a fun one.  If we get to rename this would we call it "node-ipam-controller" or "pod-ipam-controller" or "pod-range-ipam-controller" or eve "node-pod-range-ipam-controller" ?

@aojea @sarveshr7 @khenidak

Reply:
Let's see:
- it mains goal is to populate the node.Spec.podCIDR field, 
- the legacy name was "nodeipam"

I don't know, `node-ipam-controller` sounds less disruptive ...  but I definitely think it should keep "node" and "ipam" on the name

Reply:
It doesn't really do node-ipam (IPAM for nodes) or pod-ipam (IPAM for pods).  It allocated pod ranges for nodes. "node-pod-range-allocator-controller".

s/allocator/ipam -> "node-pod-range-ipam-controller"

s/range/cidr -> "node-pod-cidr-ipam-controller" ?

IPAM doesn't quite fit, how about "node-pod-cidr-controller" ?

Reply:
ping @aojea :)

Reply:
sorry, didn't see this message until now